packages/protect/__tests__/json-protect.test.ts

```diff
@@ -0,0 +1,1217 @@
+import 'dotenv/config'
+import { csColumn, csTable, csValue } from '@cipherstash/schema'
+import { beforeAll, describe, expect, it } from 'vitest'
+import { LockContext, protect } from '../src'
+
+const users = csTable('users', {
+  email: csColumn('email').freeTextSearch().equality().orderAndRange(),
+  address: csColumn('address').freeTextSearch(),
+  json: csColumn('json').dataType('jsonb').searchableJson('users/json'),
+  metadata: {
+    profile: csValue('metadata.profile'),
+    settings: {
+      preferences: csValue('metadata.settings.preferences'),
+    },
+  },
+})
+
+type User = {
+  id: string
+  email?: string | null
+  createdAt?: Date
+  updatedAt?: Date
+  address?: string | null
+  json?: Record<string, unknown> | null
```

Comment:
I think the `| null` is redundant because you have made the field itself optional.